{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "# !pip install torch\n",
    "# !pip install tqdm\n",
    "# !pip install transformers\n",
    "# !pip install scikit-learn\n",
    "# !pip install pandas\n",
    "# !pip install wandb\n",
    "# !pip install evaluate\n",
    "# !pip install transformers[torch]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/leesk/miniconda3/lib/python3.12/site-packages/tqdm/auto.py:21: TqdmWarning: IProgress not found. Please update jupyter and ipywidgets. See https://ipywidgets.readthedocs.io/en/stable/user_install.html\n",
      "  from .autonotebook import tqdm as notebook_tqdm\n"
     ]
    }
   ],
   "source": [
    "import pandas as pd\n",
    "import pickle\n",
    "from transformers import AutoTokenizer, AutoModelForSequenceClassification, TrainingArguments, Trainer, DataCollatorWithPadding, AutoConfig\n",
    "import evaluate\n",
    "import numpy as np\n",
    "from tqdm import tqdm\n",
    "from sklearn.model_selection import train_test_split\n",
    "import torch\n",
    "from sklearn.metrics import f1_score, accuracy_score\n",
    "from sklearn.metrics import classification_report\n",
    "from customModel import customBertForSequenceClassification, customRobertaForSequenceClassification,customGPT2ForSequenceClassification, customElectraForSequenceClassification\n",
    "from CustomTraniner import CustomTrainer, CustomTrainer_cross_entropy\n",
    "from transformers.configuration_utils import PretrainedConfig\n",
    "import wandb\n",
    "import random\n",
    "from transformers.models.auto.modeling_auto import MODEL_FOR_SEQUENCE_CLASSIFICATION_MAPPING\n",
    "from frontModelCustom import frontModelDataset, data_labels\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "cuda\n"
     ]
    }
   ],
   "source": [
    "device = torch.device('cuda' if torch.cuda.is_available() else 'cpu')\n",
    "print(device)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [],
   "source": [
    "def set_seed(seed:int = 42):\n",
    "    torch.manual_seed(seed)\n",
    "    torch.cuda.manual_seed(seed)\n",
    "    torch.cuda.manual_seed_all(seed)  # if use multi-GPU\n",
    "    torch.backends.cudnn.deterministic = True\n",
    "    torch.backends.cudnn.benchmark = False\n",
    "    np.random.seed(seed)\n",
    "    random.seed(seed)\n",
    "set_seed(42)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Data Process"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>singer</th>\n",
       "      <th>song_name</th>\n",
       "      <th>index</th>\n",
       "      <th>emotion</th>\n",
       "      <th>caption</th>\n",
       "      <th>tempo</th>\n",
       "      <th>instruments</th>\n",
       "      <th>tempo(int)</th>\n",
       "      <th>tempo(category)</th>\n",
       "      <th>genre</th>\n",
       "      <th>sub_genre</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>10,000 Maniacs</td>\n",
       "      <td>A Campfire Song</td>\n",
       "      <td>2</td>\n",
       "      <td>nostalgia</td>\n",
       "      <td>The melody is mellow and soothing, with a gent...</td>\n",
       "      <td>Tempo_135.0</td>\n",
       "      <td>['Program_-1', 'Program_16', 'Program_24', 'Pr...</td>\n",
       "      <td>135</td>\n",
       "      <td>Allegro</td>\n",
       "      <td>Rock</td>\n",
       "      <td>Alternative rock</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>101 Strings</td>\n",
       "      <td>Theme From The Godfather</td>\n",
       "      <td>3</td>\n",
       "      <td>nostalgia</td>\n",
       "      <td>Elegant and sweeping orchestral melody with a ...</td>\n",
       "      <td>Tempo_80.0</td>\n",
       "      <td>['Program_-1', 'Program_0', 'Program_12', 'Pro...</td>\n",
       "      <td>80</td>\n",
       "      <td>Moderato</td>\n",
       "      <td>Pop</td>\n",
       "      <td>Pop</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>10cc</td>\n",
       "      <td>Dreadlock Holiday</td>\n",
       "      <td>4</td>\n",
       "      <td>excitement</td>\n",
       "      <td>catchy and upbeat, featuring a memorable guita...</td>\n",
       "      <td>Tempo_103.0</td>\n",
       "      <td>['Program_-1', 'Program_0']</td>\n",
       "      <td>103</td>\n",
       "      <td>Moderato</td>\n",
       "      <td>Rock</td>\n",
       "      <td>Reggae rock</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>10cc</td>\n",
       "      <td>I'm Not In Love</td>\n",
       "      <td>5</td>\n",
       "      <td>nostalgia</td>\n",
       "      <td>A catchy and memorable tune with a simple yet ...</td>\n",
       "      <td>Tempo_65.0</td>\n",
       "      <td>['Program_-1', 'Program_0', 'Program_122', 'Pr...</td>\n",
       "      <td>65</td>\n",
       "      <td>Andante</td>\n",
       "      <td>Rock</td>\n",
       "      <td>Soft rock</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>10cc</td>\n",
       "      <td>The Things We Do for Love</td>\n",
       "      <td>6</td>\n",
       "      <td>love</td>\n",
       "      <td>The melody is catchy and memorable, with a ble...</td>\n",
       "      <td>Tempo_111.0</td>\n",
       "      <td>['Program_-1', 'Program_0', 'Program_16', 'Pro...</td>\n",
       "      <td>111</td>\n",
       "      <td>Moderato</td>\n",
       "      <td>Rock</td>\n",
       "      <td>Soft rock</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "           singer                  song_name  index     emotion  \\\n",
       "0  10,000 Maniacs            A Campfire Song      2   nostalgia   \n",
       "1     101 Strings   Theme From The Godfather      3   nostalgia   \n",
       "2            10cc          Dreadlock Holiday      4  excitement   \n",
       "3            10cc            I'm Not In Love      5   nostalgia   \n",
       "4            10cc  The Things We Do for Love      6        love   \n",
       "\n",
       "                                             caption        tempo  \\\n",
       "0  The melody is mellow and soothing, with a gent...  Tempo_135.0   \n",
       "1  Elegant and sweeping orchestral melody with a ...   Tempo_80.0   \n",
       "2  catchy and upbeat, featuring a memorable guita...  Tempo_103.0   \n",
       "3  A catchy and memorable tune with a simple yet ...   Tempo_65.0   \n",
       "4  The melody is catchy and memorable, with a ble...  Tempo_111.0   \n",
       "\n",
       "                                         instruments  tempo(int)  \\\n",
       "0  ['Program_-1', 'Program_16', 'Program_24', 'Pr...         135   \n",
       "1  ['Program_-1', 'Program_0', 'Program_12', 'Pro...          80   \n",
       "2                        ['Program_-1', 'Program_0']         103   \n",
       "3  ['Program_-1', 'Program_0', 'Program_122', 'Pr...          65   \n",
       "4  ['Program_-1', 'Program_0', 'Program_16', 'Pro...         111   \n",
       "\n",
       "  tempo(category) genre         sub_genre  \n",
       "0         Allegro  Rock  Alternative rock  \n",
       "1        Moderato   Pop               Pop  \n",
       "2        Moderato  Rock       Reggae rock  \n",
       "3         Andante  Rock         Soft rock  \n",
       "4        Moderato  Rock         Soft rock  "
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "data = pd.read_csv('./genre_11_tempo_4.csv')\n",
    "data.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [],
   "source": [
    "with open('labels.pkl','wb') as f:\n",
    "    pickle.dump(data.emotion.unique(),f)\n",
    "    pickle.dump(data['tempo(category)'].unique(),f)\n",
    "    pickle.dump(data['genre'].unique(),f)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [],
   "source": [
    "# with open('labels.pkl','rb') as f:\n",
    "#     pickle.dump(data.emotion.unique(),f)\n",
    "#     pickle.dump(data['tempo(category)'].unique(),f)\n",
    "#     pickle.dump(data['genre'].unique(),f)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [],
   "source": [
    "# id2label_emotion = {k:l for k, l in enumerate(data.emotion.unique())}\n",
    "# label2id_emotion = {l:k for k, l in enumerate(data.emotion.unique())}\n",
    "# id2label_tempo = {k:l for k, l in enumerate(data['tempo(category)'].unique())}\n",
    "# label2id_tempo = {l:k for k, l in enumerate(data['tempo(category)'].unique())}\n",
    "# id2label_genre = {k:l for k, l in enumerate(data['genre'].unique())}\n",
    "# label2id_genre = {l:k for k, l in enumerate(data['genre'].unique())}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [],
   "source": [
    "# train_data, valid_data = train_test_split(data, stratify=data['emotion'],test_size= 0.1, random_state=42)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [],
   "source": [
    "class customDataset_after():\n",
    "    def __init__(self, data, tokenizer):\n",
    "        self.tokenizer = tokenizer\n",
    "        self.dataset = []\n",
    "        datas = []\n",
    "        self.labels1 = []\n",
    "        self.labels2 = []\n",
    "        self.labels3 = []\n",
    "        for idx, df in tqdm(data.iterrows()):\n",
    "            label1 = [0. for _ in range(len(id2label_emotion))]\n",
    "            label2 = [0. for _ in range(len(id2label_tempo))]\n",
    "            label3 = [0. for _ in range(len(id2label_genre))]\n",
    "            datas.append(df.caption)\n",
    "            label1[label2id_emotion[df.emotion]] = 1.\n",
    "            label2[label2id_tempo[df['tempo(category)']]] = 1.\n",
    "            label3[label2id_genre[df['genre']]] = 1.\n",
    "            self.labels1.append(label1)\n",
    "            self.labels2.append(label2)\n",
    "            self.labels3.append(label3)\n",
    "        self.dataset =  tokenizer(datas,padding=True, truncation=True,max_length=512 ,return_tensors=\"pt\").to('cuda')\n",
    "        self.labels1= torch.tensor(self.labels1)\n",
    "        self.labels2= torch.tensor(self.labels2)\n",
    "        self.labels3= torch.tensor(self.labels3)\n",
    "    def __len__(self):\n",
    "        return len(self.labels1)\n",
    "    \n",
    "    def __getitem__(self, idx):\n",
    "        item = {key: val[idx].clone().detach() for key, val in self.dataset.items()}\n",
    "        item['labels1'] = self.labels1[idx].clone().detach()\n",
    "        item['labels2'] = self.labels2[idx].clone().detach()\n",
    "        item['labels3'] = self.labels3[idx].clone().detach()\n",
    "        return item"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [],
   "source": [
    "emotion , tempo, genre = data_labels('labels.pkl')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Some weights of customRobertaForSequenceClassification were not initialized from the model checkpoint at FacebookAI/roberta-large and are newly initialized: ['classifier1.bias', 'classifier1.weight', 'classifier2.bias', 'classifier2.weight', 'classifier3.bias', 'classifier3.weight', 'dense.bias', 'dense.weight']\n",
      "You should probably TRAIN this model on a down-stream task to be able to use it for predictions and inference.\n"
     ]
    }
   ],
   "source": [
    "BASE_MODEL = 'FacebookAI/roberta-large'\n",
    "\n",
    "tokenizer = AutoTokenizer.from_pretrained(BASE_MODEL)\n",
    "config = AutoConfig.from_pretrained(BASE_MODEL)\n",
    "config.num_labels1 = len(emotion)\n",
    "config.num_labels2 = len(tempo)\n",
    "config.num_labels3 = len(genre)\n",
    "# model = customBertForSequenceClassification.from_pretrained(BASE_MODEL, config= config).to(device)\n",
    "model = customRobertaForSequenceClassification.from_pretrained(BASE_MODEL, config= config).to(device)\n",
    "# model = customElectraForSequenceClassification.from_pretrained(BASE_MODEL, config= config).to(device)\n",
    "# model = customGPT2ForSequenceClassification.from_pretrained(BASE_MODEL, config= config).to(device)\n",
    "# tokenizer.pad_token = tokenizer.eos_token\n",
    "# model.config.pad_token_id = model.config.eos_token_id"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "transformers.models.roberta.configuration_roberta.RobertaConfig"
      ]
     },
     "execution_count": 13,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "type(config)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [],
   "source": [
    "data2 = data.copy()\n",
    "emotion_data = data2.groupby('emotion').sample(frac=0.05, random_state=42)\n",
    "tempo_data = data2.groupby('tempo(category)').sample(frac=0.05, random_state=42)\n",
    "genre_data = data2.groupby('genre').sample(frac=0.05, random_state=42)\n",
    "index_total = set(emotion_data.index) | set(tempo_data.index) | set(genre_data.index)\n",
    "valid_data = data2.iloc[list(index_total)]\n",
    "train_data = data2.drop(list(index_total)).sample(frac=1, random_state=42)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "8552it [00:00, 17437.99it/s]\n",
      "1419it [00:00, 17077.02it/s]\n"
     ]
    }
   ],
   "source": [
    "# dataset_train = frontModelDataset(train_data, tokenizer =tokenizer)\n",
    "# dataset_valid = frontModelDataset(valid_data, tokenizer =tokenizer)\n",
    "\n",
    "dataset_train = frontModelDataset(train_data, tokenizer =tokenizer)\n",
    "dataset_valid = frontModelDataset(valid_data, tokenizer =tokenizer)\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [],
   "source": [
    "GLOBAL_SCORE_INDICES = range(0,17)\n",
    "CAUSE_INDICES = range(17, 25)\n",
    "def get_preds_from_logits(logits):\n",
    "    ret = np.zeros(logits.shape)\n",
    "    \n",
    "    # The first 5 columns (GLOBAL_SCORE_INDICES) are for global scores. They should be handled with a multiclass approach\n",
    "    # i.e. we fill 1 to the class with highest probability, and 0 into the other columns\n",
    "    best_class = np.argmax(logits, axis=1)\n",
    "    ret[list(range(len(ret))), best_class] = 1\n",
    "    # The other columns are for causes and emotions. They should be handled with multilabel approach.\n",
    "    # i.e. we fill 1 to every class whose score is higher than some threshold\n",
    "    # In this example, we choose that threshold = 0\n",
    "    \n",
    "    return ret"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [],
   "source": [
    "def compute_metrics(eval_pred):\n",
    "    logits, labels = eval_pred\n",
    "    final_metrics = {}\n",
    "    \n",
    "    # Deduce predictions from logits\n",
    "    predictions_emotion = get_preds_from_logits(logits[0])\n",
    "    predictions_tempo = get_preds_from_logits(logits[1])\n",
    "    predictions_genre = get_preds_from_logits(logits[2])\n",
    "    \n",
    "    # Get f1 metrics for global scoring. Notice that f1_micro = accuracy\n",
    "    final_metrics[\"f1_emotion\"] = f1_score(labels[0], predictions_emotion, average=\"micro\")\n",
    "    \n",
    "    # Get f1 metrics for causes\n",
    "    final_metrics[\"f1_tempo\"] = f1_score(labels[1], predictions_tempo, average=\"micro\")\n",
    "    \n",
    "\n",
    "    # The global f1_metrics\n",
    "    final_metrics[\"f1_genre\"] = f1_score(labels[2], predictions_genre, average=\"micro\")\n",
    "\n",
    "    final_metrics['fi_total'] = (final_metrics[\"f1_emotion\"] + final_metrics[\"f1_tempo\"] + final_metrics[\"f1_genre\"])/3\n",
    "    \n",
    "    # Classification report\n",
    "    # print(\"Classification report for global scores: \")\n",
    "    # print(classification_report(labels[:, GLOBAL_SCORE_INDICES], predictions[:, GLOBAL_SCORE_INDICES], zero_division=0))\n",
    "    # print(\"Classification report for causes: \")\n",
    "    # print(classification_report(labels[:, CAUSE_INDICES], predictions[:, CAUSE_INDICES], zero_division=0))\n",
    "    return final_metrics"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [],
   "source": [
    "data_collator = DataCollatorWithPadding(tokenizer=tokenizer)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Failed to detect the name of this notebook, you can set it manually with the WANDB_NOTEBOOK_NAME environment variable to enable code saving.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "10 4 11\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\u001b[34m\u001b[1mwandb\u001b[0m: Currently logged in as: \u001b[33mleesk9663\u001b[0m (\u001b[33msanggang\u001b[0m). Use \u001b[1m`wandb login --relogin`\u001b[0m to force relogin\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "Tracking run with wandb version 0.16.4"
      ],
      "text/plain": [
       "<IPython.core.display.HTML object>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "Run data is saved locally in <code>/home/leesk/LEVEL3/wandb/run-20240313_154937-g706gojr</code>"
      ],
      "text/plain": [
       "<IPython.core.display.HTML object>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "Syncing run <strong><a href='https://wandb.ai/sanggang/Final%20project/runs/g706gojr' target=\"_blank\">cross_entropyFacebookAI/roberta-large</a></strong> to <a href='https://wandb.ai/sanggang/Final%20project' target=\"_blank\">Weights & Biases</a> (<a href='https://wandb.me/run' target=\"_blank\">docs</a>)<br/>"
      ],
      "text/plain": [
       "<IPython.core.display.HTML object>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       " View project at <a href='https://wandb.ai/sanggang/Final%20project' target=\"_blank\">https://wandb.ai/sanggang/Final%20project</a>"
      ],
      "text/plain": [
       "<IPython.core.display.HTML object>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       " View run at <a href='https://wandb.ai/sanggang/Final%20project/runs/g706gojr' target=\"_blank\">https://wandb.ai/sanggang/Final%20project/runs/g706gojr</a>"
      ],
      "text/plain": [
       "<IPython.core.display.HTML object>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Detected kernel version 5.4.0, which is below the recommended minimum of 5.5.0; this can cause the process to hang. It is recommended to upgrade the kernel to the minimum version or higher.\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "\n",
       "    <div>\n",
       "      \n",
       "      <progress value='2401' max='3210' style='width:300px; height:20px; vertical-align: middle;'></progress>\n",
       "      [2401/3210 26:25 < 08:54, 1.51 it/s, Epoch 4.49/6]\n",
       "    </div>\n",
       "    <table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       " <tr style=\"text-align: left;\">\n",
       "      <th>Step</th>\n",
       "      <th>Training Loss</th>\n",
       "      <th>Validation Loss</th>\n",
       "      <th>F1 Emotion</th>\n",
       "      <th>F1 Tempo</th>\n",
       "      <th>F1 Genre</th>\n",
       "      <th>Fi Total</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>300</td>\n",
       "      <td>4.287500</td>\n",
       "      <td>1.291284</td>\n",
       "      <td>0.537703</td>\n",
       "      <td>0.382664</td>\n",
       "      <td>0.415786</td>\n",
       "      <td>0.445384</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>600</td>\n",
       "      <td>4.196100</td>\n",
       "      <td>1.242160</td>\n",
       "      <td>0.582100</td>\n",
       "      <td>0.362932</td>\n",
       "      <td>0.477097</td>\n",
       "      <td>0.474043</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>900</td>\n",
       "      <td>4.030300</td>\n",
       "      <td>1.202066</td>\n",
       "      <td>0.585624</td>\n",
       "      <td>0.411557</td>\n",
       "      <td>0.495419</td>\n",
       "      <td>0.497533</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1200</td>\n",
       "      <td>3.841300</td>\n",
       "      <td>1.193132</td>\n",
       "      <td>0.606061</td>\n",
       "      <td>0.379140</td>\n",
       "      <td>0.490486</td>\n",
       "      <td>0.491896</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1500</td>\n",
       "      <td>3.848100</td>\n",
       "      <td>1.121353</td>\n",
       "      <td>0.606061</td>\n",
       "      <td>0.388302</td>\n",
       "      <td>0.489782</td>\n",
       "      <td>0.494715</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1800</td>\n",
       "      <td>3.921200</td>\n",
       "      <td>1.126720</td>\n",
       "      <td>0.625088</td>\n",
       "      <td>0.400282</td>\n",
       "      <td>0.518675</td>\n",
       "      <td>0.514682</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2100</td>\n",
       "      <td>3.687000</td>\n",
       "      <td>1.126656</td>\n",
       "      <td>0.608175</td>\n",
       "      <td>0.373502</td>\n",
       "      <td>0.513037</td>\n",
       "      <td>0.498238</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2400</td>\n",
       "      <td>3.537900</td>\n",
       "      <td>1.133826</td>\n",
       "      <td>0.602537</td>\n",
       "      <td>0.415786</td>\n",
       "      <td>0.520085</td>\n",
       "      <td>0.512802</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table><p>"
      ],
      "text/plain": [
       "<IPython.core.display.HTML object>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "ename": "RuntimeError",
     "evalue": "[enforce fail at inline_container.cc:595] . unexpected pos 1804376256 vs 1804376148",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mRuntimeError\u001b[0m                              Traceback (most recent call last)",
      "File \u001b[0;32m~/miniconda3/lib/python3.12/site-packages/torch/serialization.py:629\u001b[0m, in \u001b[0;36msave\u001b[0;34m(obj, f, pickle_module, pickle_protocol, _use_new_zipfile_serialization, _disable_byteorder_record)\u001b[0m\n\u001b[1;32m    628\u001b[0m \u001b[38;5;28;01mwith\u001b[39;00m _open_zipfile_writer(f) \u001b[38;5;28;01mas\u001b[39;00m opened_zipfile:\n\u001b[0;32m--> 629\u001b[0m     \u001b[43m_save\u001b[49m\u001b[43m(\u001b[49m\u001b[43mobj\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mopened_zipfile\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mpickle_module\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mpickle_protocol\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43m_disable_byteorder_record\u001b[49m\u001b[43m)\u001b[49m\n\u001b[1;32m    630\u001b[0m     \u001b[38;5;28;01mreturn\u001b[39;00m\n",
      "File \u001b[0;32m~/miniconda3/lib/python3.12/site-packages/torch/serialization.py:863\u001b[0m, in \u001b[0;36m_save\u001b[0;34m(obj, zip_file, pickle_module, pickle_protocol, _disable_byteorder_record)\u001b[0m\n\u001b[1;32m    862\u001b[0m num_bytes \u001b[38;5;241m=\u001b[39m storage\u001b[38;5;241m.\u001b[39mnbytes()\n\u001b[0;32m--> 863\u001b[0m \u001b[43mzip_file\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mwrite_record\u001b[49m\u001b[43m(\u001b[49m\u001b[43mname\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mstorage\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mdata_ptr\u001b[49m\u001b[43m(\u001b[49m\u001b[43m)\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mnum_bytes\u001b[49m\u001b[43m)\u001b[49m\n",
      "\u001b[0;31mRuntimeError\u001b[0m: [enforce fail at inline_container.cc:764] . PytorchStreamWriter failed writing file data/4: file write failed",
      "\nDuring handling of the above exception, another exception occurred:\n",
      "\u001b[0;31mRuntimeError\u001b[0m                              Traceback (most recent call last)",
      "Cell \u001b[0;32mIn[19], line 60\u001b[0m\n\u001b[1;32m     37\u001b[0m trainer \u001b[38;5;241m=\u001b[39m CustomTrainer_cross_entropy(\n\u001b[1;32m     38\u001b[0m    model\u001b[38;5;241m=\u001b[39mmodel,\n\u001b[1;32m     39\u001b[0m    args\u001b[38;5;241m=\u001b[39mtraining_args,\n\u001b[0;32m   (...)\u001b[0m\n\u001b[1;32m     44\u001b[0m    compute_metrics\u001b[38;5;241m=\u001b[39mcompute_metrics,\n\u001b[1;32m     45\u001b[0m )\n\u001b[1;32m     47\u001b[0m \u001b[38;5;66;03m# trainer = Trainer(\u001b[39;00m\n\u001b[1;32m     48\u001b[0m \n\u001b[1;32m     49\u001b[0m \u001b[38;5;66;03m#    model=model,\u001b[39;00m\n\u001b[0;32m   (...)\u001b[0m\n\u001b[1;32m     55\u001b[0m \u001b[38;5;66;03m#    compute_metrics=compute_metrics,\u001b[39;00m\n\u001b[1;32m     56\u001b[0m \u001b[38;5;66;03m# )\u001b[39;00m\n\u001b[0;32m---> 60\u001b[0m \u001b[43mtrainer\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mtrain\u001b[49m\u001b[43m(\u001b[49m\u001b[43m)\u001b[49m\n",
      "File \u001b[0;32m~/miniconda3/lib/python3.12/site-packages/transformers/trainer.py:1624\u001b[0m, in \u001b[0;36mTrainer.train\u001b[0;34m(self, resume_from_checkpoint, trial, ignore_keys_for_eval, **kwargs)\u001b[0m\n\u001b[1;32m   1622\u001b[0m         hf_hub_utils\u001b[38;5;241m.\u001b[39menable_progress_bars()\n\u001b[1;32m   1623\u001b[0m \u001b[38;5;28;01melse\u001b[39;00m:\n\u001b[0;32m-> 1624\u001b[0m     \u001b[38;5;28;01mreturn\u001b[39;00m \u001b[43minner_training_loop\u001b[49m\u001b[43m(\u001b[49m\n\u001b[1;32m   1625\u001b[0m \u001b[43m        \u001b[49m\u001b[43margs\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43margs\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m   1626\u001b[0m \u001b[43m        \u001b[49m\u001b[43mresume_from_checkpoint\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mresume_from_checkpoint\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m   1627\u001b[0m \u001b[43m        \u001b[49m\u001b[43mtrial\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mtrial\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m   1628\u001b[0m \u001b[43m        \u001b[49m\u001b[43mignore_keys_for_eval\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mignore_keys_for_eval\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m   1629\u001b[0m \u001b[43m    \u001b[49m\u001b[43m)\u001b[49m\n",
      "File \u001b[0;32m~/miniconda3/lib/python3.12/site-packages/transformers/trainer.py:2029\u001b[0m, in \u001b[0;36mTrainer._inner_training_loop\u001b[0;34m(self, batch_size, args, resume_from_checkpoint, trial, ignore_keys_for_eval)\u001b[0m\n\u001b[1;32m   2026\u001b[0m     \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39mstate\u001b[38;5;241m.\u001b[39mepoch \u001b[38;5;241m=\u001b[39m epoch \u001b[38;5;241m+\u001b[39m (step \u001b[38;5;241m+\u001b[39m \u001b[38;5;241m1\u001b[39m \u001b[38;5;241m+\u001b[39m steps_skipped) \u001b[38;5;241m/\u001b[39m steps_in_epoch\n\u001b[1;32m   2027\u001b[0m     \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39mcontrol \u001b[38;5;241m=\u001b[39m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39mcallback_handler\u001b[38;5;241m.\u001b[39mon_step_end(args, \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39mstate, \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39mcontrol)\n\u001b[0;32m-> 2029\u001b[0m     \u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43m_maybe_log_save_evaluate\u001b[49m\u001b[43m(\u001b[49m\u001b[43mtr_loss\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mgrad_norm\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mmodel\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mtrial\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mepoch\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mignore_keys_for_eval\u001b[49m\u001b[43m)\u001b[49m\n\u001b[1;32m   2030\u001b[0m \u001b[38;5;28;01melse\u001b[39;00m:\n\u001b[1;32m   2031\u001b[0m     \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39mcontrol \u001b[38;5;241m=\u001b[39m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39mcallback_handler\u001b[38;5;241m.\u001b[39mon_substep_end(args, \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39mstate, \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39mcontrol)\n",
      "File \u001b[0;32m~/miniconda3/lib/python3.12/site-packages/transformers/trainer.py:2423\u001b[0m, in \u001b[0;36mTrainer._maybe_log_save_evaluate\u001b[0;34m(self, tr_loss, grad_norm, model, trial, epoch, ignore_keys_for_eval)\u001b[0m\n\u001b[1;32m   2420\u001b[0m         \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39mlr_scheduler\u001b[38;5;241m.\u001b[39mstep(metrics[metric_to_check])\n\u001b[1;32m   2422\u001b[0m \u001b[38;5;28;01mif\u001b[39;00m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39mcontrol\u001b[38;5;241m.\u001b[39mshould_save:\n\u001b[0;32m-> 2423\u001b[0m     \u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43m_save_checkpoint\u001b[49m\u001b[43m(\u001b[49m\u001b[43mmodel\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mtrial\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mmetrics\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mmetrics\u001b[49m\u001b[43m)\u001b[49m\n\u001b[1;32m   2424\u001b[0m     \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39mcontrol \u001b[38;5;241m=\u001b[39m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39mcallback_handler\u001b[38;5;241m.\u001b[39mon_save(\u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39margs, \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39mstate, \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39mcontrol)\n",
      "File \u001b[0;32m~/miniconda3/lib/python3.12/site-packages/transformers/trainer.py:2503\u001b[0m, in \u001b[0;36mTrainer._save_checkpoint\u001b[0;34m(self, model, trial, metrics)\u001b[0m\n\u001b[1;32m   2499\u001b[0m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39msave_model(staging_output_dir, _internal_call\u001b[38;5;241m=\u001b[39m\u001b[38;5;28;01mTrue\u001b[39;00m)\n\u001b[1;32m   2501\u001b[0m \u001b[38;5;28;01mif\u001b[39;00m \u001b[38;5;129;01mnot\u001b[39;00m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39margs\u001b[38;5;241m.\u001b[39msave_only_model:\n\u001b[1;32m   2502\u001b[0m     \u001b[38;5;66;03m# Save optimizer and scheduler\u001b[39;00m\n\u001b[0;32m-> 2503\u001b[0m     \u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43m_save_optimizer_and_scheduler\u001b[49m\u001b[43m(\u001b[49m\u001b[43mstaging_output_dir\u001b[49m\u001b[43m)\u001b[49m\n\u001b[1;32m   2504\u001b[0m     \u001b[38;5;66;03m# Save RNG state\u001b[39;00m\n\u001b[1;32m   2505\u001b[0m     \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39m_save_rng_state(staging_output_dir)\n",
      "File \u001b[0;32m~/miniconda3/lib/python3.12/site-packages/transformers/trainer.py:2628\u001b[0m, in \u001b[0;36mTrainer._save_optimizer_and_scheduler\u001b[0;34m(self, output_dir)\u001b[0m\n\u001b[1;32m   2623\u001b[0m     save_fsdp_optimizer(\n\u001b[1;32m   2624\u001b[0m         \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39maccelerator\u001b[38;5;241m.\u001b[39mstate\u001b[38;5;241m.\u001b[39mfsdp_plugin, \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39maccelerator, \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39moptimizer, \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39mmodel, output_dir\n\u001b[1;32m   2625\u001b[0m     )\n\u001b[1;32m   2626\u001b[0m \u001b[38;5;28;01melif\u001b[39;00m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39margs\u001b[38;5;241m.\u001b[39mshould_save:\n\u001b[1;32m   2627\u001b[0m     \u001b[38;5;66;03m# deepspeed.save_checkpoint above saves model/optim/sched\u001b[39;00m\n\u001b[0;32m-> 2628\u001b[0m     \u001b[43mtorch\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43msave\u001b[49m\u001b[43m(\u001b[49m\u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43moptimizer\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mstate_dict\u001b[49m\u001b[43m(\u001b[49m\u001b[43m)\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mos\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mpath\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mjoin\u001b[49m\u001b[43m(\u001b[49m\u001b[43moutput_dir\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mOPTIMIZER_NAME\u001b[49m\u001b[43m)\u001b[49m\u001b[43m)\u001b[49m\n\u001b[1;32m   2630\u001b[0m \u001b[38;5;66;03m# Save SCHEDULER & SCALER\u001b[39;00m\n\u001b[1;32m   2631\u001b[0m is_deepspeed_custom_scheduler \u001b[38;5;241m=\u001b[39m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39mis_deepspeed_enabled \u001b[38;5;129;01mand\u001b[39;00m \u001b[38;5;129;01mnot\u001b[39;00m \u001b[38;5;28misinstance\u001b[39m(\n\u001b[1;32m   2632\u001b[0m     \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39mlr_scheduler, DeepSpeedSchedulerWrapper\n\u001b[1;32m   2633\u001b[0m )\n",
      "File \u001b[0;32m~/miniconda3/lib/python3.12/site-packages/torch/serialization.py:628\u001b[0m, in \u001b[0;36msave\u001b[0;34m(obj, f, pickle_module, pickle_protocol, _use_new_zipfile_serialization, _disable_byteorder_record)\u001b[0m\n\u001b[1;32m    625\u001b[0m _check_save_filelike(f)\n\u001b[1;32m    627\u001b[0m \u001b[38;5;28;01mif\u001b[39;00m _use_new_zipfile_serialization:\n\u001b[0;32m--> 628\u001b[0m \u001b[43m    \u001b[49m\u001b[38;5;28;43;01mwith\u001b[39;49;00m\u001b[43m \u001b[49m\u001b[43m_open_zipfile_writer\u001b[49m\u001b[43m(\u001b[49m\u001b[43mf\u001b[49m\u001b[43m)\u001b[49m\u001b[43m \u001b[49m\u001b[38;5;28;43;01mas\u001b[39;49;00m\u001b[43m \u001b[49m\u001b[43mopened_zipfile\u001b[49m\u001b[43m:\u001b[49m\n\u001b[1;32m    629\u001b[0m \u001b[43m        \u001b[49m\u001b[43m_save\u001b[49m\u001b[43m(\u001b[49m\u001b[43mobj\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mopened_zipfile\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mpickle_module\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mpickle_protocol\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43m_disable_byteorder_record\u001b[49m\u001b[43m)\u001b[49m\n\u001b[1;32m    630\u001b[0m \u001b[43m        \u001b[49m\u001b[38;5;28;43;01mreturn\u001b[39;49;00m\n",
      "File \u001b[0;32m~/miniconda3/lib/python3.12/site-packages/torch/serialization.py:476\u001b[0m, in \u001b[0;36m_open_zipfile_writer_file.__exit__\u001b[0;34m(self, *args)\u001b[0m\n\u001b[1;32m    475\u001b[0m \u001b[38;5;28;01mdef\u001b[39;00m \u001b[38;5;21m__exit__\u001b[39m(\u001b[38;5;28mself\u001b[39m, \u001b[38;5;241m*\u001b[39margs) \u001b[38;5;241m-\u001b[39m\u001b[38;5;241m>\u001b[39m \u001b[38;5;28;01mNone\u001b[39;00m:\n\u001b[0;32m--> 476\u001b[0m     \u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mfile_like\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mwrite_end_of_file\u001b[49m\u001b[43m(\u001b[49m\u001b[43m)\u001b[49m\n\u001b[1;32m    477\u001b[0m     \u001b[38;5;28;01mif\u001b[39;00m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39mfile_stream \u001b[38;5;129;01mis\u001b[39;00m \u001b[38;5;129;01mnot\u001b[39;00m \u001b[38;5;28;01mNone\u001b[39;00m:\n\u001b[1;32m    478\u001b[0m         \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39mfile_stream\u001b[38;5;241m.\u001b[39mclose()\n",
      "\u001b[0;31mRuntimeError\u001b[0m: [enforce fail at inline_container.cc:595] . unexpected pos 1804376256 vs 1804376148"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "--- Logging error ---\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/leesk/miniconda3/lib/python3.12/logging/__init__.py\", line 1164, in emit\n",
      "    self.flush()\n",
      "  File \"/home/leesk/miniconda3/lib/python3.12/logging/__init__.py\", line 1144, in flush\n",
      "    self.stream.flush()\n",
      "OSError: [Errno 28] No space left on device\n",
      "Call stack:\n",
      "  File \"/home/leesk/miniconda3/lib/python3.12/threading.py\", line 1030, in _bootstrap\n",
      "    self._bootstrap_inner()\n",
      "  File \"/home/leesk/miniconda3/lib/python3.12/threading.py\", line 1073, in _bootstrap_inner\n",
      "    self.run()\n",
      "  File \"/home/leesk/miniconda3/lib/python3.12/site-packages/wandb/sdk/internal/internal_util.py\", line 49, in run\n",
      "    self._run()\n",
      "  File \"/home/leesk/miniconda3/lib/python3.12/site-packages/wandb/sdk/internal/internal_util.py\", line 100, in _run\n",
      "    self._process(record)\n",
      "  File \"/home/leesk/miniconda3/lib/python3.12/site-packages/wandb/sdk/internal/internal.py\", line 279, in _process\n",
      "    self._hm.handle(record)\n",
      "  File \"/home/leesk/miniconda3/lib/python3.12/site-packages/wandb/sdk/internal/handler.py\", line 138, in handle\n",
      "    handler(record)\n",
      "  File \"/home/leesk/miniconda3/lib/python3.12/site-packages/wandb/sdk/internal/handler.py\", line 148, in handle_request\n",
      "    handler(record)\n",
      "  File \"/home/leesk/miniconda3/lib/python3.12/site-packages/wandb/sdk/internal/handler.py\", line 709, in handle_request_pause\n",
      "    self._system_monitor.finish()\n",
      "  File \"/home/leesk/miniconda3/lib/python3.12/site-packages/wandb/sdk/internal/system/system_monitor.py\", line 203, in finish\n",
      "    logger.info(\"Stopping system monitor\")\n",
      "Message: 'Stopping system monitor'\n",
      "Arguments: ()\n",
      "--- Logging error ---\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/leesk/miniconda3/lib/python3.12/logging/__init__.py\", line 1164, in emit\n",
      "    self.flush()\n",
      "  File \"/home/leesk/miniconda3/lib/python3.12/logging/__init__.py\", line 1144, in flush\n",
      "    self.stream.flush()\n",
      "OSError: [Errno 28] No space left on device\n",
      "Call stack:\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "  File \"/home/leesk/miniconda3/lib/python3.12/threading.py\", line 1030, in _bootstrap\n",
      "    self._bootstrap_inner()\n",
      "  File \"/home/leesk/miniconda3/lib/python3.12/threading.py\", line 1073, in _bootstrap_inner\n",
      "    self.run()\n",
      "  File \"/home/leesk/miniconda3/lib/python3.12/threading.py\", line 1010, in run\n",
      "    self._target(*self._args, **self._kwargs)\n",
      "  File \"/home/leesk/miniconda3/lib/python3.12/site-packages/wandb/sdk/internal/system/system_monitor.py\", line 179, in _start\n",
      "    logger.debug(\"Finished system metrics aggregation loop\")\n",
      "Message: 'Finished system metrics aggregation loop'\n",
      "Arguments: ()\n",
      "--- Logging error ---\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/leesk/miniconda3/lib/python3.12/logging/__init__.py\", line 1164, in emit\n",
      "    self.flush()\n",
      "  File \"/home/leesk/miniconda3/lib/python3.12/logging/__init__.py\", line 1144, in flush\n",
      "    self.stream.flush()\n",
      "OSError: [Errno 28] No space left on device\n",
      "Call stack:\n",
      "  File \"/home/leesk/miniconda3/lib/python3.12/threading.py\", line 1030, in _bootstrap\n",
      "    self._bootstrap_inner()\n",
      "  File \"/home/leesk/miniconda3/lib/python3.12/threading.py\", line 1073, in _bootstrap_inner\n",
      "    self.run()\n",
      "  File \"/home/leesk/miniconda3/lib/python3.12/site-packages/wandb/sdk/internal/internal_util.py\", line 49, in run\n",
      "    self._run()\n",
      "  File \"/home/leesk/miniconda3/lib/python3.12/site-packages/wandb/sdk/internal/internal_util.py\", line 100, in _run\n",
      "    self._process(record)\n",
      "  File \"/home/leesk/miniconda3/lib/python3.12/site-packages/wandb/sdk/internal/internal.py\", line 279, in _process\n",
      "    self._hm.handle(record)\n",
      "  File \"/home/leesk/miniconda3/lib/python3.12/site-packages/wandb/sdk/internal/handler.py\", line 138, in handle\n",
      "    handler(record)\n",
      "  File \"/home/leesk/miniconda3/lib/python3.12/site-packages/wandb/sdk/internal/handler.py\", line 148, in handle_request\n",
      "    handler(record)\n",
      "  File \"/home/leesk/miniconda3/lib/python3.12/site-packages/wandb/sdk/internal/handler.py\", line 709, in handle_request_pause\n",
      "    self._system_monitor.finish()\n",
      "  File \"/home/leesk/miniconda3/lib/python3.12/site-packages/wandb/sdk/internal/system/system_monitor.py\", line 206, in finish\n",
      "    asset.finish()\n",
      "  File \"/home/leesk/miniconda3/lib/python3.12/site-packages/wandb/sdk/internal/system/assets/cpu.py\", line 163, in finish\n",
      "    self.metrics_monitor.finish()\n",
      "  File \"/home/leesk/miniconda3/lib/python3.12/site-packages/wandb/sdk/internal/system/assets/interfaces.py\", line 202, in finish\n",
      "    logger.info(f\"Joined {thread_name} monitor\")\n",
      "Message: 'Joined cpu monitor'\n",
      "Arguments: ()\n",
      "--- Logging error ---\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/leesk/miniconda3/lib/python3.12/logging/__init__.py\", line 1164, in emit\n",
      "    self.flush()\n",
      "  File \"/home/leesk/miniconda3/lib/python3.12/logging/__init__.py\", line 1144, in flush\n",
      "    self.stream.flush()\n",
      "OSError: [Errno 28] No space left on device\n",
      "Call stack:\n",
      "  File \"/home/leesk/miniconda3/lib/python3.12/threading.py\", line 1030, in _bootstrap\n",
      "    self._bootstrap_inner()\n",
      "  File \"/home/leesk/miniconda3/lib/python3.12/threading.py\", line 1073, in _bootstrap_inner\n",
      "    self.run()\n",
      "  File \"/home/leesk/miniconda3/lib/python3.12/threading.py\", line 1010, in run\n",
      "    self._target(*self._args, **self._kwargs)\n",
      "  File \"/home/leesk/miniconda3/lib/python3.12/site-packages/wandb/sdk/internal/system/system_monitor.py\", line 183, in _start\n",
      "    logger.debug(\"Publishing last batch of metrics\")\n",
      "Message: 'Publishing last batch of metrics'\n",
      "Arguments: ()\n",
      "--- Logging error ---\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/leesk/miniconda3/lib/python3.12/logging/__init__.py\", line 1164, in emit\n",
      "    self.flush()\n",
      "  File \"/home/leesk/miniconda3/lib/python3.12/logging/__init__.py\", line 1144, in flush\n",
      "    self.stream.flush()\n",
      "OSError: [Errno 28] No space left on device\n",
      "Call stack:\n",
      "  File \"/home/leesk/miniconda3/lib/python3.12/threading.py\", line 1030, in _bootstrap\n",
      "    self._bootstrap_inner()\n",
      "  File \"/home/leesk/miniconda3/lib/python3.12/threading.py\", line 1073, in _bootstrap_inner\n",
      "    self.run()\n",
      "  File \"/home/leesk/miniconda3/lib/python3.12/site-packages/wandb/sdk/internal/internal_util.py\", line 49, in run\n",
      "    self._run()\n",
      "  File \"/home/leesk/miniconda3/lib/python3.12/site-packages/wandb/sdk/internal/internal_util.py\", line 100, in _run\n",
      "    self._process(record)\n",
      "  File \"/home/leesk/miniconda3/lib/python3.12/site-packages/wandb/sdk/internal/internal.py\", line 279, in _process\n",
      "    self._hm.handle(record)\n",
      "  File \"/home/leesk/miniconda3/lib/python3.12/site-packages/wandb/sdk/internal/handler.py\", line 138, in handle\n",
      "    handler(record)\n",
      "  File \"/home/leesk/miniconda3/lib/python3.12/site-packages/wandb/sdk/internal/handler.py\", line 148, in handle_request\n",
      "    handler(record)\n",
      "  File \"/home/leesk/miniconda3/lib/python3.12/site-packages/wandb/sdk/internal/handler.py\", line 709, in handle_request_pause\n",
      "    self._system_monitor.finish()\n",
      "  File \"/home/leesk/miniconda3/lib/python3.12/site-packages/wandb/sdk/internal/system/system_monitor.py\", line 206, in finish\n",
      "    asset.finish()\n",
      "  File \"/home/leesk/miniconda3/lib/python3.12/site-packages/wandb/sdk/internal/system/assets/disk.py\", line 210, in finish\n",
      "    self.metrics_monitor.finish()\n",
      "  File \"/home/leesk/miniconda3/lib/python3.12/site-packages/wandb/sdk/internal/system/assets/interfaces.py\", line 202, in finish\n",
      "    logger.info(f\"Joined {thread_name} monitor\")\n",
      "Message: 'Joined disk monitor'\n",
      "Arguments: ()\n",
      "--- Logging error ---\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/leesk/miniconda3/lib/python3.12/logging/__init__.py\", line 1164, in emit\n",
      "    self.flush()\n",
      "  File \"/home/leesk/miniconda3/lib/python3.12/logging/__init__.py\", line 1144, in flush\n",
      "    self.stream.flush()\n",
      "OSError: [Errno 28] No space left on device\n",
      "Call stack:\n",
      "  File \"/home/leesk/miniconda3/lib/python3.12/threading.py\", line 1030, in _bootstrap\n",
      "    self._bootstrap_inner()\n",
      "  File \"/home/leesk/miniconda3/lib/python3.12/threading.py\", line 1073, in _bootstrap_inner\n",
      "    self.run()\n",
      "  File \"/home/leesk/miniconda3/lib/python3.12/site-packages/wandb/sdk/internal/internal_util.py\", line 49, in run\n",
      "    self._run()\n",
      "  File \"/home/leesk/miniconda3/lib/python3.12/site-packages/wandb/sdk/internal/internal_util.py\", line 100, in _run\n",
      "    self._process(record)\n",
      "  File \"/home/leesk/miniconda3/lib/python3.12/site-packages/wandb/sdk/internal/internal.py\", line 279, in _process\n",
      "    self._hm.handle(record)\n",
      "  File \"/home/leesk/miniconda3/lib/python3.12/site-packages/wandb/sdk/internal/handler.py\", line 138, in handle\n",
      "    handler(record)\n",
      "  File \"/home/leesk/miniconda3/lib/python3.12/site-packages/wandb/sdk/internal/handler.py\", line 148, in handle_request\n",
      "    handler(record)\n",
      "  File \"/home/leesk/miniconda3/lib/python3.12/site-packages/wandb/sdk/internal/handler.py\", line 709, in handle_request_pause\n",
      "    self._system_monitor.finish()\n",
      "  File \"/home/leesk/miniconda3/lib/python3.12/site-packages/wandb/sdk/internal/system/system_monitor.py\", line 206, in finish\n",
      "    asset.finish()\n",
      "  File \"/home/leesk/miniconda3/lib/python3.12/site-packages/wandb/sdk/internal/system/assets/gpu.py\", line 388, in finish\n",
      "    self.metrics_monitor.finish()\n",
      "  File \"/home/leesk/miniconda3/lib/python3.12/site-packages/wandb/sdk/internal/system/assets/interfaces.py\", line 202, in finish\n",
      "    logger.info(f\"Joined {thread_name} monitor\")\n",
      "Message: 'Joined gpu monitor'\n",
      "Arguments: ()\n",
      "--- Logging error ---\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/leesk/miniconda3/lib/python3.12/logging/__init__.py\", line 1164, in emit\n",
      "    self.flush()\n",
      "  File \"/home/leesk/miniconda3/lib/python3.12/logging/__init__.py\", line 1144, in flush\n",
      "    self.stream.flush()\n",
      "OSError: [Errno 28] No space left on device\n",
      "Call stack:\n",
      "  File \"/home/leesk/miniconda3/lib/python3.12/threading.py\", line 1030, in _bootstrap\n",
      "    self._bootstrap_inner()\n",
      "  File \"/home/leesk/miniconda3/lib/python3.12/threading.py\", line 1073, in _bootstrap_inner\n",
      "    self.run()\n",
      "  File \"/home/leesk/miniconda3/lib/python3.12/site-packages/wandb/sdk/internal/internal_util.py\", line 49, in run\n",
      "    self._run()\n",
      "  File \"/home/leesk/miniconda3/lib/python3.12/site-packages/wandb/sdk/internal/internal_util.py\", line 100, in _run\n",
      "    self._process(record)\n",
      "  File \"/home/leesk/miniconda3/lib/python3.12/site-packages/wandb/sdk/internal/internal.py\", line 279, in _process\n",
      "    self._hm.handle(record)\n",
      "  File \"/home/leesk/miniconda3/lib/python3.12/site-packages/wandb/sdk/internal/handler.py\", line 138, in handle\n",
      "    handler(record)\n",
      "  File \"/home/leesk/miniconda3/lib/python3.12/site-packages/wandb/sdk/internal/handler.py\", line 148, in handle_request\n",
      "    handler(record)\n",
      "  File \"/home/leesk/miniconda3/lib/python3.12/site-packages/wandb/sdk/internal/handler.py\", line 709, in handle_request_pause\n",
      "    self._system_monitor.finish()\n",
      "  File \"/home/leesk/miniconda3/lib/python3.12/site-packages/wandb/sdk/internal/system/system_monitor.py\", line 206, in finish\n",
      "    asset.finish()\n",
      "  File \"/home/leesk/miniconda3/lib/python3.12/site-packages/wandb/sdk/internal/system/assets/memory.py\", line 152, in finish\n",
      "    self.metrics_monitor.finish()\n",
      "  File \"/home/leesk/miniconda3/lib/python3.12/site-packages/wandb/sdk/internal/system/assets/interfaces.py\", line 202, in finish\n",
      "    logger.info(f\"Joined {thread_name} monitor\")\n",
      "Message: 'Joined memory monitor'\n",
      "Arguments: ()\n",
      "--- Logging error ---\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/leesk/miniconda3/lib/python3.12/logging/__init__.py\", line 1164, in emit\n",
      "    self.flush()\n",
      "  File \"/home/leesk/miniconda3/lib/python3.12/logging/__init__.py\", line 1144, in flush\n",
      "    self.stream.flush()\n",
      "OSError: [Errno 28] No space left on device\n",
      "Call stack:\n",
      "  File \"/home/leesk/miniconda3/lib/python3.12/threading.py\", line 1030, in _bootstrap\n",
      "    self._bootstrap_inner()\n",
      "  File \"/home/leesk/miniconda3/lib/python3.12/threading.py\", line 1073, in _bootstrap_inner\n",
      "    self.run()\n",
      "  File \"/home/leesk/miniconda3/lib/python3.12/site-packages/wandb/sdk/internal/internal_util.py\", line 49, in run\n",
      "    self._run()\n",
      "  File \"/home/leesk/miniconda3/lib/python3.12/site-packages/wandb/sdk/internal/internal_util.py\", line 100, in _run\n",
      "    self._process(record)\n",
      "  File \"/home/leesk/miniconda3/lib/python3.12/site-packages/wandb/sdk/internal/internal.py\", line 279, in _process\n",
      "    self._hm.handle(record)\n",
      "  File \"/home/leesk/miniconda3/lib/python3.12/site-packages/wandb/sdk/internal/handler.py\", line 138, in handle\n",
      "    handler(record)\n",
      "  File \"/home/leesk/miniconda3/lib/python3.12/site-packages/wandb/sdk/internal/handler.py\", line 148, in handle_request\n",
      "    handler(record)\n",
      "  File \"/home/leesk/miniconda3/lib/python3.12/site-packages/wandb/sdk/internal/handler.py\", line 709, in handle_request_pause\n",
      "    self._system_monitor.finish()\n",
      "  File \"/home/leesk/miniconda3/lib/python3.12/site-packages/wandb/sdk/internal/system/system_monitor.py\", line 206, in finish\n",
      "    asset.finish()\n",
      "  File \"/home/leesk/miniconda3/lib/python3.12/site-packages/wandb/sdk/internal/system/assets/network.py\", line 96, in finish\n",
      "    self.metrics_monitor.finish()\n",
      "  File \"/home/leesk/miniconda3/lib/python3.12/site-packages/wandb/sdk/internal/system/assets/interfaces.py\", line 202, in finish\n",
      "    logger.info(f\"Joined {thread_name} monitor\")\n",
      "Message: 'Joined network monitor'\n",
      "Arguments: ()\n",
      "--- Logging error ---\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/leesk/miniconda3/lib/python3.12/logging/__init__.py\", line 1164, in emit\n",
      "    self.flush()\n",
      "  File \"/home/leesk/miniconda3/lib/python3.12/logging/__init__.py\", line 1144, in flush\n",
      "    self.stream.flush()\n",
      "OSError: [Errno 28] No space left on device\n",
      "Call stack:\n",
      "  File \"/home/leesk/miniconda3/lib/python3.12/threading.py\", line 1030, in _bootstrap\n",
      "    self._bootstrap_inner()\n",
      "  File \"/home/leesk/miniconda3/lib/python3.12/threading.py\", line 1073, in _bootstrap_inner\n",
      "    self.run()\n",
      "  File \"/home/leesk/miniconda3/lib/python3.12/site-packages/wandb/sdk/internal/internal_util.py\", line 49, in run\n",
      "    self._run()\n",
      "  File \"/home/leesk/miniconda3/lib/python3.12/site-packages/wandb/sdk/internal/internal_util.py\", line 101, in _run\n",
      "    self._finish()\n",
      "  File \"/home/leesk/miniconda3/lib/python3.12/site-packages/wandb/sdk/internal/internal.py\", line 282, in _finish\n",
      "    self._hm.finish()\n",
      "  File \"/home/leesk/miniconda3/lib/python3.12/site-packages/wandb/sdk/internal/handler.py\", line 869, in finish\n",
      "    logger.info(\"shutting down handler\")\n",
      "Message: 'shutting down handler'\n",
      "Arguments: ()\n",
      "--- Logging error ---\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/leesk/miniconda3/lib/python3.12/logging/__init__.py\", line 1164, in emit\n",
      "    self.flush()\n",
      "  File \"/home/leesk/miniconda3/lib/python3.12/logging/__init__.py\", line 1144, in flush\n",
      "    self.stream.flush()\n",
      "OSError: [Errno 28] No space left on device\n",
      "Call stack:\n",
      "  File \"/home/leesk/miniconda3/lib/python3.12/threading.py\", line 1030, in _bootstrap\n",
      "    self._bootstrap_inner()\n",
      "  File \"/home/leesk/miniconda3/lib/python3.12/threading.py\", line 1073, in _bootstrap_inner\n",
      "    self.run()\n",
      "  File \"/home/leesk/miniconda3/lib/python3.12/site-packages/wandb/sdk/internal/internal_util.py\", line 49, in run\n",
      "    self._run()\n",
      "  File \"/home/leesk/miniconda3/lib/python3.12/site-packages/wandb/sdk/internal/internal_util.py\", line 101, in _run\n",
      "    self._finish()\n",
      "  File \"/home/leesk/miniconda3/lib/python3.12/site-packages/wandb/sdk/internal/internal.py\", line 331, in _finish\n",
      "    self._sm.finish()\n",
      "  File \"/home/leesk/miniconda3/lib/python3.12/site-packages/wandb/sdk/internal/sender.py\", line 1521, in finish\n",
      "    logger.info(\"shutting down sender\")\n",
      "Message: 'shutting down sender'\n",
      "Arguments: ()\n",
      "--- Logging error ---\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/leesk/miniconda3/lib/python3.12/logging/__init__.py\", line 1164, in emit\n",
      "    self.flush()\n",
      "  File \"/home/leesk/miniconda3/lib/python3.12/logging/__init__.py\", line 1144, in flush\n",
      "    self.stream.flush()\n",
      "OSError: [Errno 28] No space left on device\n",
      "Call stack:\n",
      "  File \"/home/leesk/miniconda3/lib/python3.12/threading.py\", line 1030, in _bootstrap\n",
      "    self._bootstrap_inner()\n",
      "  File \"/home/leesk/miniconda3/lib/python3.12/threading.py\", line 1073, in _bootstrap_inner\n",
      "    self.run()\n",
      "  File \"/home/leesk/miniconda3/lib/python3.12/site-packages/wandb/sdk/internal/internal_util.py\", line 49, in run\n",
      "    self._run()\n",
      "  File \"/home/leesk/miniconda3/lib/python3.12/site-packages/wandb/sdk/internal/internal_util.py\", line 101, in _run\n",
      "    self._finish()\n",
      "  File \"/home/leesk/miniconda3/lib/python3.12/site-packages/wandb/sdk/internal/internal.py\", line 331, in _finish\n",
      "    self._sm.finish()\n",
      "  File \"/home/leesk/miniconda3/lib/python3.12/site-packages/wandb/sdk/internal/sender.py\", line 1526, in finish\n",
      "    self._dir_watcher.finish()\n",
      "  File \"/home/leesk/miniconda3/lib/python3.12/site-packages/wandb/filesync/dir_watcher.py\", line 358, in finish\n",
      "    logger.info(\"shutting down directory watcher\")\n",
      "Message: 'shutting down directory watcher'\n",
      "Arguments: ()\n",
      "--- Logging error ---\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/leesk/miniconda3/lib/python3.12/logging/__init__.py\", line 1164, in emit\n",
      "    self.flush()\n",
      "  File \"/home/leesk/miniconda3/lib/python3.12/logging/__init__.py\", line 1144, in flush\n",
      "    self.stream.flush()\n",
      "OSError: [Errno 28] No space left on device\n",
      "Call stack:\n",
      "  File \"/home/leesk/miniconda3/lib/python3.12/threading.py\", line 1030, in _bootstrap\n",
      "    self._bootstrap_inner()\n",
      "  File \"/home/leesk/miniconda3/lib/python3.12/threading.py\", line 1073, in _bootstrap_inner\n",
      "    self.run()\n",
      "  File \"/home/leesk/miniconda3/lib/python3.12/site-packages/wandb/sdk/internal/internal_util.py\", line 49, in run\n",
      "    self._run()\n",
      "  File \"/home/leesk/miniconda3/lib/python3.12/site-packages/wandb/sdk/internal/internal_util.py\", line 101, in _run\n",
      "    self._finish()\n",
      "  File \"/home/leesk/miniconda3/lib/python3.12/site-packages/wandb/sdk/internal/internal.py\", line 331, in _finish\n",
      "    self._sm.finish()\n",
      "  File \"/home/leesk/miniconda3/lib/python3.12/site-packages/wandb/sdk/internal/sender.py\", line 1526, in finish\n",
      "    self._dir_watcher.finish()\n",
      "  File \"/home/leesk/miniconda3/lib/python3.12/site-packages/wandb/filesync/dir_watcher.py\", line 372, in finish\n",
      "    self._file_observer.dispatch_events(\n",
      "  File \"/home/leesk/miniconda3/lib/python3.12/site-packages/wandb/vendor/watchdog_0_9_0/wandb_watchdog/observers/api.py\", line 368, in dispatch_events\n",
      "    handler.dispatch(event)\n",
      "  File \"/home/leesk/miniconda3/lib/python3.12/site-packages/wandb/vendor/watchdog_0_9_0/wandb_watchdog/events.py\", line 454, in dispatch\n",
      "    _method_map[event_type](event)\n",
      "  File \"/home/leesk/miniconda3/lib/python3.12/site-packages/wandb/filesync/dir_watcher.py\", line 288, in _on_file_modified\n",
      "    logger.info(f\"file/dir modified: { event.src_path}\")\n",
      "Message: 'file/dir modified: /home/leesk/LEVEL3/wandb/run-20240313_154937-g706gojr/files/output.log'\n",
      "Arguments: ()\n",
      "--- Logging error ---\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/leesk/miniconda3/lib/python3.12/logging/__init__.py\", line 1164, in emit\n",
      "    self.flush()\n",
      "  File \"/home/leesk/miniconda3/lib/python3.12/logging/__init__.py\", line 1144, in flush\n",
      "    self.stream.flush()\n",
      "OSError: [Errno 28] No space left on device\n",
      "Call stack:\n",
      "  File \"/home/leesk/miniconda3/lib/python3.12/threading.py\", line 1030, in _bootstrap\n",
      "    self._bootstrap_inner()\n",
      "  File \"/home/leesk/miniconda3/lib/python3.12/threading.py\", line 1073, in _bootstrap_inner\n",
      "    self.run()\n",
      "  File \"/home/leesk/miniconda3/lib/python3.12/site-packages/wandb/sdk/internal/internal_util.py\", line 49, in run\n",
      "    self._run()\n",
      "  File \"/home/leesk/miniconda3/lib/python3.12/site-packages/wandb/sdk/internal/internal_util.py\", line 101, in _run\n",
      "    self._finish()\n",
      "  File \"/home/leesk/miniconda3/lib/python3.12/site-packages/wandb/sdk/internal/internal.py\", line 331, in _finish\n",
      "    self._sm.finish()\n",
      "  File \"/home/leesk/miniconda3/lib/python3.12/site-packages/wandb/sdk/internal/sender.py\", line 1526, in finish\n",
      "    self._dir_watcher.finish()\n",
      "  File \"/home/leesk/miniconda3/lib/python3.12/site-packages/wandb/filesync/dir_watcher.py\", line 388, in finish\n",
      "    logger.info(\"scan: %s\", self._dir)\n",
      "Message: 'scan: %s'\n",
      "Arguments: ('/home/leesk/LEVEL3/wandb/run-20240313_154937-g706gojr/files',)\n",
      "--- Logging error ---\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/leesk/miniconda3/lib/python3.12/logging/__init__.py\", line 1164, in emit\n",
      "    self.flush()\n",
      "  File \"/home/leesk/miniconda3/lib/python3.12/logging/__init__.py\", line 1144, in flush\n",
      "    self.stream.flush()\n",
      "OSError: [Errno 28] No space left on device\n",
      "Call stack:\n",
      "  File \"/home/leesk/miniconda3/lib/python3.12/threading.py\", line 1030, in _bootstrap\n",
      "    self._bootstrap_inner()\n",
      "  File \"/home/leesk/miniconda3/lib/python3.12/threading.py\", line 1073, in _bootstrap_inner\n",
      "    self.run()\n",
      "  File \"/home/leesk/miniconda3/lib/python3.12/site-packages/wandb/sdk/internal/internal_util.py\", line 49, in run\n",
      "    self._run()\n",
      "  File \"/home/leesk/miniconda3/lib/python3.12/site-packages/wandb/sdk/internal/internal_util.py\", line 101, in _run\n",
      "    self._finish()\n",
      "  File \"/home/leesk/miniconda3/lib/python3.12/site-packages/wandb/sdk/internal/internal.py\", line 331, in _finish\n",
      "    self._sm.finish()\n",
      "  File \"/home/leesk/miniconda3/lib/python3.12/site-packages/wandb/sdk/internal/sender.py\", line 1526, in finish\n",
      "    self._dir_watcher.finish()\n",
      "  File \"/home/leesk/miniconda3/lib/python3.12/site-packages/wandb/filesync/dir_watcher.py\", line 402, in finish\n",
      "    logger.info(\"scan save: %s %s\", file_path, save_name)\n",
      "Message: 'scan save: %s %s'\n",
      "Arguments: ('/home/leesk/LEVEL3/wandb/run-20240313_154937-g706gojr/files/requirements.txt', 'requirements.txt')\n",
      "--- Logging error ---\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/leesk/miniconda3/lib/python3.12/logging/__init__.py\", line 1164, in emit\n",
      "    self.flush()\n",
      "  File \"/home/leesk/miniconda3/lib/python3.12/logging/__init__.py\", line 1144, in flush\n",
      "    self.stream.flush()\n",
      "OSError: [Errno 28] No space left on device\n",
      "Call stack:\n",
      "  File \"/home/leesk/miniconda3/lib/python3.12/threading.py\", line 1030, in _bootstrap\n",
      "    self._bootstrap_inner()\n",
      "  File \"/home/leesk/miniconda3/lib/python3.12/threading.py\", line 1073, in _bootstrap_inner\n",
      "    self.run()\n",
      "  File \"/home/leesk/miniconda3/lib/python3.12/site-packages/wandb/sdk/internal/internal_util.py\", line 49, in run\n",
      "    self._run()\n",
      "  File \"/home/leesk/miniconda3/lib/python3.12/site-packages/wandb/sdk/internal/internal_util.py\", line 101, in _run\n",
      "    self._finish()\n",
      "  File \"/home/leesk/miniconda3/lib/python3.12/site-packages/wandb/sdk/internal/internal.py\", line 331, in _finish\n",
      "    self._sm.finish()\n",
      "  File \"/home/leesk/miniconda3/lib/python3.12/site-packages/wandb/sdk/internal/sender.py\", line 1526, in finish\n",
      "    self._dir_watcher.finish()\n",
      "  File \"/home/leesk/miniconda3/lib/python3.12/site-packages/wandb/filesync/dir_watcher.py\", line 402, in finish\n",
      "    logger.info(\"scan save: %s %s\", file_path, save_name)\n",
      "Message: 'scan save: %s %s'\n",
      "Arguments: ('/home/leesk/LEVEL3/wandb/run-20240313_154937-g706gojr/files/wandb-summary.json', 'wandb-summary.json')\n",
      "--- Logging error ---\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/leesk/miniconda3/lib/python3.12/logging/__init__.py\", line 1164, in emit\n",
      "    self.flush()\n",
      "  File \"/home/leesk/miniconda3/lib/python3.12/logging/__init__.py\", line 1144, in flush\n",
      "    self.stream.flush()\n",
      "OSError: [Errno 28] No space left on device\n",
      "Call stack:\n",
      "  File \"/home/leesk/miniconda3/lib/python3.12/threading.py\", line 1030, in _bootstrap\n",
      "    self._bootstrap_inner()\n",
      "  File \"/home/leesk/miniconda3/lib/python3.12/threading.py\", line 1073, in _bootstrap_inner\n",
      "    self.run()\n",
      "  File \"/home/leesk/miniconda3/lib/python3.12/site-packages/wandb/sdk/internal/internal_util.py\", line 49, in run\n",
      "    self._run()\n",
      "  File \"/home/leesk/miniconda3/lib/python3.12/site-packages/wandb/sdk/internal/internal_util.py\", line 101, in _run\n",
      "    self._finish()\n",
      "  File \"/home/leesk/miniconda3/lib/python3.12/site-packages/wandb/sdk/internal/internal.py\", line 331, in _finish\n",
      "    self._sm.finish()\n",
      "  File \"/home/leesk/miniconda3/lib/python3.12/site-packages/wandb/sdk/internal/sender.py\", line 1526, in finish\n",
      "    self._dir_watcher.finish()\n",
      "  File \"/home/leesk/miniconda3/lib/python3.12/site-packages/wandb/filesync/dir_watcher.py\", line 402, in finish\n",
      "    logger.info(\"scan save: %s %s\", file_path, save_name)\n",
      "Message: 'scan save: %s %s'\n",
      "Arguments: ('/home/leesk/LEVEL3/wandb/run-20240313_154937-g706gojr/files/wandb-metadata.json', 'wandb-metadata.json')\n",
      "--- Logging error ---\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/leesk/miniconda3/lib/python3.12/logging/__init__.py\", line 1164, in emit\n",
      "    self.flush()\n",
      "  File \"/home/leesk/miniconda3/lib/python3.12/logging/__init__.py\", line 1144, in flush\n",
      "    self.stream.flush()\n",
      "OSError: [Errno 28] No space left on device\n",
      "Call stack:\n",
      "  File \"/home/leesk/miniconda3/lib/python3.12/threading.py\", line 1030, in _bootstrap\n",
      "    self._bootstrap_inner()\n",
      "  File \"/home/leesk/miniconda3/lib/python3.12/threading.py\", line 1073, in _bootstrap_inner\n",
      "    self.run()\n",
      "  File \"/home/leesk/miniconda3/lib/python3.12/site-packages/wandb/sdk/internal/internal_util.py\", line 49, in run\n",
      "    self._run()\n",
      "  File \"/home/leesk/miniconda3/lib/python3.12/site-packages/wandb/sdk/internal/internal_util.py\", line 101, in _run\n",
      "    self._finish()\n",
      "  File \"/home/leesk/miniconda3/lib/python3.12/site-packages/wandb/sdk/internal/internal.py\", line 331, in _finish\n",
      "    self._sm.finish()\n",
      "  File \"/home/leesk/miniconda3/lib/python3.12/site-packages/wandb/sdk/internal/sender.py\", line 1526, in finish\n",
      "    self._dir_watcher.finish()\n",
      "  File \"/home/leesk/miniconda3/lib/python3.12/site-packages/wandb/filesync/dir_watcher.py\", line 402, in finish\n",
      "    logger.info(\"scan save: %s %s\", file_path, save_name)\n",
      "Message: 'scan save: %s %s'\n",
      "Arguments: ('/home/leesk/LEVEL3/wandb/run-20240313_154937-g706gojr/files/conda-environment.yaml', 'conda-environment.yaml')\n",
      "--- Logging error ---\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/leesk/miniconda3/lib/python3.12/logging/__init__.py\", line 1164, in emit\n",
      "    self.flush()\n",
      "  File \"/home/leesk/miniconda3/lib/python3.12/logging/__init__.py\", line 1144, in flush\n",
      "    self.stream.flush()\n",
      "OSError: [Errno 28] No space left on device\n",
      "Call stack:\n",
      "  File \"/home/leesk/miniconda3/lib/python3.12/threading.py\", line 1030, in _bootstrap\n",
      "    self._bootstrap_inner()\n",
      "  File \"/home/leesk/miniconda3/lib/python3.12/threading.py\", line 1073, in _bootstrap_inner\n",
      "    self.run()\n",
      "  File \"/home/leesk/miniconda3/lib/python3.12/site-packages/wandb/sdk/internal/internal_util.py\", line 49, in run\n",
      "    self._run()\n",
      "  File \"/home/leesk/miniconda3/lib/python3.12/site-packages/wandb/sdk/internal/internal_util.py\", line 101, in _run\n",
      "    self._finish()\n",
      "  File \"/home/leesk/miniconda3/lib/python3.12/site-packages/wandb/sdk/internal/internal.py\", line 331, in _finish\n",
      "    self._sm.finish()\n",
      "  File \"/home/leesk/miniconda3/lib/python3.12/site-packages/wandb/sdk/internal/sender.py\", line 1526, in finish\n",
      "    self._dir_watcher.finish()\n",
      "  File \"/home/leesk/miniconda3/lib/python3.12/site-packages/wandb/filesync/dir_watcher.py\", line 402, in finish\n",
      "    logger.info(\"scan save: %s %s\", file_path, save_name)\n",
      "Message: 'scan save: %s %s'\n",
      "Arguments: ('/home/leesk/LEVEL3/wandb/run-20240313_154937-g706gojr/files/output.log', 'output.log')\n",
      "--- Logging error ---\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/leesk/miniconda3/lib/python3.12/logging/__init__.py\", line 1164, in emit\n",
      "    self.flush()\n",
      "  File \"/home/leesk/miniconda3/lib/python3.12/logging/__init__.py\", line 1144, in flush\n",
      "    self.stream.flush()\n",
      "OSError: [Errno 28] No space left on device\n",
      "Call stack:\n",
      "  File \"/home/leesk/miniconda3/lib/python3.12/threading.py\", line 1030, in _bootstrap\n",
      "    self._bootstrap_inner()\n",
      "  File \"/home/leesk/miniconda3/lib/python3.12/threading.py\", line 1073, in _bootstrap_inner\n",
      "    self.run()\n",
      "  File \"/home/leesk/miniconda3/lib/python3.12/site-packages/wandb/sdk/internal/internal_util.py\", line 49, in run\n",
      "    self._run()\n",
      "  File \"/home/leesk/miniconda3/lib/python3.12/site-packages/wandb/sdk/internal/internal_util.py\", line 101, in _run\n",
      "    self._finish()\n",
      "  File \"/home/leesk/miniconda3/lib/python3.12/site-packages/wandb/sdk/internal/internal.py\", line 331, in _finish\n",
      "    self._sm.finish()\n",
      "  File \"/home/leesk/miniconda3/lib/python3.12/site-packages/wandb/sdk/internal/sender.py\", line 1526, in finish\n",
      "    self._dir_watcher.finish()\n",
      "  File \"/home/leesk/miniconda3/lib/python3.12/site-packages/wandb/filesync/dir_watcher.py\", line 402, in finish\n",
      "    logger.info(\"scan save: %s %s\", file_path, save_name)\n",
      "Message: 'scan save: %s %s'\n",
      "Arguments: ('/home/leesk/LEVEL3/wandb/run-20240313_154937-g706gojr/files/config.yaml', 'config.yaml')\n",
      "--- Logging error ---\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/leesk/miniconda3/lib/python3.12/logging/__init__.py\", line 1164, in emit\n",
      "    self.flush()\n",
      "  File \"/home/leesk/miniconda3/lib/python3.12/logging/__init__.py\", line 1144, in flush\n",
      "    self.stream.flush()\n",
      "OSError: [Errno 28] No space left on device\n",
      "Call stack:\n",
      "  File \"/home/leesk/miniconda3/lib/python3.12/threading.py\", line 1030, in _bootstrap\n",
      "    self._bootstrap_inner()\n",
      "  File \"/home/leesk/miniconda3/lib/python3.12/threading.py\", line 1073, in _bootstrap_inner\n",
      "    self.run()\n",
      "  File \"/home/leesk/miniconda3/lib/python3.12/site-packages/wandb/sdk/internal/internal_util.py\", line 49, in run\n",
      "    self._run()\n",
      "  File \"/home/leesk/miniconda3/lib/python3.12/site-packages/wandb/sdk/internal/internal_util.py\", line 101, in _run\n",
      "    self._finish()\n",
      "  File \"/home/leesk/miniconda3/lib/python3.12/site-packages/wandb/sdk/internal/internal.py\", line 331, in _finish\n",
      "    self._sm.finish()\n",
      "  File \"/home/leesk/miniconda3/lib/python3.12/site-packages/wandb/sdk/internal/sender.py\", line 1529, in finish\n",
      "    self._pusher.finish()\n",
      "  File \"/home/leesk/miniconda3/lib/python3.12/site-packages/wandb/sdk/internal/file_pusher.py\", line 172, in finish\n",
      "    logger.info(\"shutting down file pusher\")\n",
      "Message: 'shutting down file pusher'\n",
      "Arguments: ()\n",
      "--- Logging error ---\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/leesk/miniconda3/lib/python3.12/logging/__init__.py\", line 1164, in emit\n",
      "    self.flush()\n",
      "  File \"/home/leesk/miniconda3/lib/python3.12/logging/__init__.py\", line 1144, in flush\n",
      "    self.stream.flush()\n",
      "OSError: [Errno 28] No space left on device\n",
      "Call stack:\n",
      "  File \"/home/leesk/miniconda3/lib/python3.12/threading.py\", line 1030, in _bootstrap\n",
      "    self._bootstrap_inner()\n",
      "  File \"/home/leesk/miniconda3/lib/python3.12/threading.py\", line 1073, in _bootstrap_inner\n",
      "    self.run()\n",
      "  File \"/home/leesk/miniconda3/lib/python3.12/site-packages/wandb/sdk/internal/internal_util.py\", line 49, in run\n",
      "    self._run()\n",
      "  File \"/home/leesk/miniconda3/lib/python3.12/site-packages/wandb/sdk/internal/internal_util.py\", line 101, in _run\n",
      "    self._finish()\n",
      "  File \"/home/leesk/miniconda3/lib/python3.12/site-packages/wandb/sdk/internal/internal.py\", line 331, in _finish\n",
      "    self._sm.finish()\n",
      "  File \"/home/leesk/miniconda3/lib/python3.12/site-packages/wandb/sdk/internal/sender.py\", line 1530, in finish\n",
      "    self._pusher.join()\n",
      "  File \"/home/leesk/miniconda3/lib/python3.12/site-packages/wandb/sdk/internal/file_pusher.py\", line 178, in join\n",
      "    logger.info(\"waiting for file pusher\")\n",
      "Message: 'waiting for file pusher'\n",
      "Arguments: ()\n",
      "Thread WriterThread:\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/leesk/miniconda3/lib/python3.12/site-packages/wandb/sdk/internal/internal_util.py\", line 49, in run\n",
      "    self._run()\n",
      "  File \"/home/leesk/miniconda3/lib/python3.12/site-packages/wandb/sdk/internal/internal_util.py\", line 100, in _run\n",
      "    self._process(record)\n",
      "  File \"/home/leesk/miniconda3/lib/python3.12/site-packages/wandb/sdk/internal/internal.py\", line 380, in _process\n",
      "    self._wm.write(record)\n",
      "  File \"/home/leesk/miniconda3/lib/python3.12/site-packages/wandb/sdk/internal/writer.py\", line 154, in write\n",
      "    write_handler(record)\n",
      "  File \"/home/leesk/miniconda3/lib/python3.12/site-packages/wandb/sdk/internal/writer.py\", line 135, in _write\n",
      "    self._write_record(record)\n",
      "  File \"/home/leesk/miniconda3/lib/python3.12/site-packages/wandb/sdk/internal/writer.py\", line 109, in _write_record\n",
      "    ret = self._ds.write(record)\n",
      "          ^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"/home/leesk/miniconda3/lib/python3.12/site-packages/wandb/sdk/internal/datastore.py\", line 291, in write\n",
      "    ret = self._write_data(s)\n",
      "          ^^^^^^^^^^^^^^^^^^^\n",
      "  File \"/home/leesk/miniconda3/lib/python3.12/site-packages/wandb/sdk/internal/datastore.py\", line 247, in _write_data\n",
      "    self._write_record(s)\n",
      "  File \"/home/leesk/miniconda3/lib/python3.12/site-packages/wandb/sdk/internal/datastore.py\", line 226, in _write_record\n",
      "    self._fp.write(s)\n",
      "OSError: [Errno 28] No space left on device\n",
      "wandb: ERROR Internal wandb error: file data was not synced\n",
      "Exception in thread NetStatThr:\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/leesk/miniconda3/lib/python3.12/threading.py\", line 1073, in _bootstrap_inner\n",
      "Exception in thread ChkStopThr:\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/leesk/miniconda3/lib/python3.12/threading.py\", line 1073, in _bootstrap_inner\n",
      "    self.run()\n",
      "  File \"/home/leesk/miniconda3/lib/python3.12/site-packages/ipykernel/ipkernel.py\", line 761, in run_closure\n",
      "Exception in thread IntMsgThr:\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/leesk/miniconda3/lib/python3.12/threading.py\", line 1073, in _bootstrap_inner\n",
      "    self.run()\n",
      "  File \"/home/leesk/miniconda3/lib/python3.12/site-packages/ipykernel/ipkernel.py\", line 761, in run_closure\n",
      "    _threading_Thread_run(self)\n",
      "  File \"/home/leesk/miniconda3/lib/python3.12/threading.py\", line 1010, in run\n",
      "    self.run()\n",
      "  File \"/home/leesk/miniconda3/lib/python3.12/site-packages/ipykernel/ipkernel.py\", line 761, in run_closure\n",
      "    self._target(*self._args, **self._kwargs)\n",
      "  File \"/home/leesk/miniconda3/lib/python3.12/site-packages/wandb/sdk/wandb_run.py\", line 268, in check_network_status\n",
      "    self._loop_check_status(\n",
      "  File \"/home/leesk/miniconda3/lib/python3.12/site-packages/wandb/sdk/wandb_run.py\", line 224, in _loop_check_status\n",
      "    local_handle = request()\n",
      "         _threading_Thread_run(self) \n",
      "   File \"/home/leesk/miniconda3/lib/python3.12/threading.py\", line 1010, in run\n",
      " _threading_Thread_run(self)     self._target(*self._args, **self._kwargs) \n",
      "   File \"/home/leesk/miniconda3/lib/python3.12/site-packages/wandb/sdk/wandb_run.py\", line 300, in check_internal_messages\n",
      " \n",
      "     self._loop_check_status( \n",
      "   File \"/home/leesk/miniconda3/lib/python3.12/site-packages/wandb/sdk/wandb_run.py\", line 224, in _loop_check_status\n",
      "      local_handle = request() \n",
      "               ^  ^  ^  ^  ^ ^^^^^^^^^^^^^\n",
      "\n",
      "  File \"/home/leesk/miniconda3/lib/python3.12/site-packages/wandb/sdk/interface/interface.py\", line 803, in deliver_internal_messages\n",
      "  File \"/home/leesk/miniconda3/lib/python3.12/site-packages/wandb/sdk/interface/interface.py\", line 795, in deliver_network_status\n",
      "  File \"/home/leesk/miniconda3/lib/python3.12/threading.py\", line 1010, in run\n",
      "        self._target(*self._args, **self._kwargs)\n",
      "return self._deliver_network_status(status)  File \"/home/leesk/miniconda3/lib/python3.12/site-packages/wandb/sdk/wandb_run.py\", line 286, in check_stop_status\n",
      "\n",
      "    self._loop_check_status(\n",
      "   File \"/home/leesk/miniconda3/lib/python3.12/site-packages/wandb/sdk/wandb_run.py\", line 224, in _loop_check_status\n",
      "      local_handle = request()\n",
      "                       ^  ^  ^^^^^^^^^^^^^^\n",
      "^  File \"/home/leesk/miniconda3/lib/python3.12/site-packages/wandb/sdk/interface/interface.py\", line 787, in deliver_stop_status\n",
      "^    ^return self._deliver_stop_status(status)\n",
      "^ ^  ^  ^  ^  ^  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "^  File \"/home/leesk/miniconda3/lib/python3.12/site-packages/wandb/sdk/interface/interface_shared.py\", line 484, in _deliver_stop_status\n",
      "^    ^return self._deliver_record(record)\n",
      "\n",
      "     File \"/home/leesk/miniconda3/lib/python3.12/site-packages/wandb/sdk/interface/interface_shared.py\", line 500, in _deliver_network_status\n",
      "     return self._deliver_internal_messages(internal_message)\n",
      "           ^^^^^^^^^^^^^^^^^^^^^ ^     ^ return self._deliver_record(record)^ \n",
      "^ ^  ^  ^^ ^^ ^^ ^^ ^^ ^^ ^^ ^^ ^^ ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "^^^^  File \"/home/leesk/miniconda3/lib/python3.12/site-packages/wandb/sdk/interface/interface_shared.py\", line 506, in _deliver_internal_messages\n",
      "^^^^^^    ^^return self._deliver_record(record)\n",
      "^\n",
      "^  File \"/home/leesk/miniconda3/lib/python3.12/site-packages/wandb/sdk/interface/interface_shared.py\", line 449, in _deliver_record\n",
      "^ ^ ^     ^ handle = mailbox._deliver_record(record, interface=self)^ \n",
      "^ ^  \n",
      "      File \"/home/leesk/miniconda3/lib/python3.12/site-packages/wandb/sdk/interface/interface_shared.py\", line 449, in _deliver_record\n",
      "    ^     ^handle = mailbox._deliver_record(record, interface=self) ^\n",
      " ^ ^  ^  ^  ^  ^ ^^ ^^ ^^ ^^ ^^ ^^ ^^ ^^ ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "^^^^  File \"/home/leesk/miniconda3/lib/python3.12/site-packages/wandb/sdk/interface/interface_shared.py\", line 449, in _deliver_record\n",
      "^^^^^^    ^^handle = mailbox._deliver_record(record, interface=self)^^\n",
      "^^^^ ^^ ^^ ^^ ^^ ^^ ^^ ^^ ^^ ^^ ^^ ^^ ^^ ^^^^^^^^^^^^^^^^^^^\n",
      "^^^^  File \"/home/leesk/miniconda3/lib/python3.12/site-packages/wandb/sdk/lib/mailbox.py\", line 455, in _deliver_record\n",
      "^^^^^    ^^interface._publish(record)^^\n",
      "^^^^  File \"/home/leesk/miniconda3/lib/python3.12/site-packages/wandb/sdk/interface/interface_sock.py\", line 51, in _publish\n",
      "\n",
      "^^  File \"/home/leesk/miniconda3/lib/python3.12/site-packages/wandb/sdk/lib/mailbox.py\", line 455, in _deliver_record\n",
      "^    ^self._sock_client.send_record_publish(record)^    \n",
      "^interface._publish(record)^\n",
      "  File \"/home/leesk/miniconda3/lib/python3.12/site-packages/wandb/sdk/lib/sock_client.py\", line 221, in send_record_publish\n",
      "^^  File \"/home/leesk/miniconda3/lib/python3.12/site-packages/wandb/sdk/interface/interface_sock.py\", line 51, in _publish\n",
      "^    ^self.send_server_request(server_req)^    \n",
      "^self._sock_client.send_record_publish(record)^\n",
      "  File \"/home/leesk/miniconda3/lib/python3.12/site-packages/wandb/sdk/lib/sock_client.py\", line 155, in send_server_request\n",
      "^^  File \"/home/leesk/miniconda3/lib/python3.12/site-packages/wandb/sdk/lib/sock_client.py\", line 221, in send_record_publish\n",
      "    ^self._send_message(msg)^\n",
      "    ^self.send_server_request(server_req)^  File \"/home/leesk/miniconda3/lib/python3.12/site-packages/wandb/sdk/lib/sock_client.py\", line 152, in _send_message\n",
      "\n",
      "^^  File \"/home/leesk/miniconda3/lib/python3.12/site-packages/wandb/sdk/lib/sock_client.py\", line 155, in send_server_request\n",
      "    ^self._sendall_with_error_handle(header + data)^\n",
      "    ^self._send_message(msg)^  File \"/home/leesk/miniconda3/lib/python3.12/site-packages/wandb/sdk/lib/sock_client.py\", line 130, in _sendall_with_error_handle\n",
      "\n",
      "^^  File \"/home/leesk/miniconda3/lib/python3.12/site-packages/wandb/sdk/lib/sock_client.py\", line 152, in _send_message\n",
      "    ^sent = self._sock.send(data)^\n",
      "    ^self._sendall_with_error_handle(header + data)^ \n",
      "\n",
      "    File \"/home/leesk/miniconda3/lib/python3.12/site-packages/wandb/sdk/lib/sock_client.py\", line 130, in _sendall_with_error_handle\n",
      "  File \"/home/leesk/miniconda3/lib/python3.12/site-packages/wandb/sdk/lib/mailbox.py\", line 455, in _deliver_record\n",
      "           sent = self._sock.send(data)interface._publish(record) \n",
      "\n",
      "     File \"/home/leesk/miniconda3/lib/python3.12/site-packages/wandb/sdk/interface/interface_sock.py\", line 51, in _publish\n",
      "        ^ self._sock_client.send_record_publish(record)^ \n",
      "^ ^   File \"/home/leesk/miniconda3/lib/python3.12/site-packages/wandb/sdk/lib/sock_client.py\", line 221, in send_record_publish\n",
      "^ ^ ^     ^self.send_server_request(server_req) ^\n",
      "^^^^  File \"/home/leesk/miniconda3/lib/python3.12/site-packages/wandb/sdk/lib/sock_client.py\", line 155, in send_server_request\n",
      "^^^^^^    ^^self._send_message(msg)^^\n",
      "^^^^  File \"/home/leesk/miniconda3/lib/python3.12/site-packages/wandb/sdk/lib/sock_client.py\", line 152, in _send_message\n",
      "^^^^^    ^^self._sendall_with_error_handle(header + data)\n",
      "^\n",
      "^BrokenPipeError^  File \"/home/leesk/miniconda3/lib/python3.12/site-packages/wandb/sdk/lib/sock_client.py\", line 130, in _sendall_with_error_handle\n",
      ": ^[Errno 32] Broken pipe^\n",
      "    ^sent = self._sock.send(data)^\n",
      "^\n",
      "  BrokenPipeError :  [Errno 32] Broken pipe \n",
      "      ^^^^^^^^^^^^^^^^^^^^^\n",
      "BrokenPipeError: [Errno 32] Broken pipe\n"
     ]
    }
   ],
   "source": [
    "print(config.num_labels1, config.num_labels2, config.num_labels3)\n",
    "\n",
    "wandb.init(project=\"Final project\", entity=\"sanggang\",name = \"cross_entropy\"+BASE_MODEL)\n",
    "\n",
    "training_args = TrainingArguments(\n",
    "\n",
    "   output_dir=\"my_awesome_model\",\n",
    "   save_steps=300,\n",
    "   eval_steps = 300, \n",
    "   warmup_steps=500,\n",
    "   logging_steps=100,\n",
    "   learning_rate=5e-5,\n",
    "   per_device_train_batch_size=16,\n",
    "   per_device_eval_batch_size=16,\n",
    "   num_train_epochs=6,\n",
    "   weight_decay=0.01,\n",
    "   evaluation_strategy='steps',\n",
    "   load_best_model_at_end = True,\n",
    "   save_total_limit = 2,\n",
    "   report_to=\"wandb\",\n",
    "   metric_for_best_model='fi_total',\n",
    "   # run_name=BASE_MODEL, \n",
    ")\n",
    "\n",
    "# trainer = CustomTrainer(\n",
    "\n",
    "#    model=model,\n",
    "#    args=training_args,\n",
    "#    train_dataset=dataset_train,\n",
    "#    eval_dataset=dataset_valid,\n",
    "#    tokenizer=tokenizer,\n",
    "#    data_collator=data_collator,\n",
    "#    compute_metrics=compute_metrics,\n",
    "# )\n",
    "\n",
    "\n",
    "trainer = CustomTrainer_cross_entropy(\n",
    "   model=model,\n",
    "   args=training_args,\n",
    "   train_dataset=dataset_train,\n",
    "   eval_dataset=dataset_valid,\n",
    "   tokenizer=tokenizer,\n",
    "   data_collator=data_collator,\n",
    "   compute_metrics=compute_metrics,\n",
    ")\n",
    "\n",
    "# trainer = Trainer(\n",
    "\n",
    "#    model=model,\n",
    "#    args=training_args,\n",
    "#    train_dataset=dataset_train,\n",
    "#    eval_dataset=dataset_valid,\n",
    "#    tokenizer=tokenizer,\n",
    "#    data_collator=data_collator,\n",
    "#    compute_metrics=compute_metrics,\n",
    "# )\n",
    "\n",
    "\n",
    "\n",
    "trainer.train()\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.12.2"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
